### PR TITLE
dev time tracker when user is idle for more than 5 mins

### PR DESCRIPTION
**Pull request by AmanPrasad43**
Thank you for your contribution.
Before submitting this PR, please make sure:

- [x] PR description and commit message should describe the changes done in this PR
- [x] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [x] Latest Code from master or specific release branch is merged to your branch
- [ ] No unwanted\commented\junk code is included
- [ ] No new warning upon build solution
- [ ] Code Summary\Comments are added to my code which explains what my code is doing
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [ ] Verify that changes are compatible with all relevant browsers and platforms.
- [ ] After creating pull request there should not be any conflicts
- [ ] Resolve all Codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Update the Help Library document to match any feature changes

## Summary by CodeRabbit

- **New Features**
  - Introduced an automated development time tracking feature that pauses tracking during user inactivity.
  - Added functionality to attach event handlers for user activity, enhancing the accuracy of development metrics.
  - Implemented a method to detach event handlers when the main window is closing.

- **Refactor**
  - Simplified the code in the main application window by encapsulating timer management within the new tracking class.
  - Improved code maintainability and readability by reducing complexity in the main window's logic.

### CODE REVIEW

File: Ginger/Ginger/DevTimeTrackerIdle.cs [hunk 1]
```diff
@@ -0,0 +1,152 @@
+﻿using Amdocs.Ginger.Repository;
```
Comment (Maheshkale447) [inline Ginger/Ginger/DevTimeTrackerIdle.cs:1]: Add license part at top of the class

File: Ginger/Ginger/DevTimeTrackerIdle.cs [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+using System;
+using System.Collections.Generic;
+using System.Windows;
+using System.Windows.Threading;
+using amdocs.ginger.GingerCoreNET;
+using Amdocs.Ginger.Common;
+
+namespace Ginger
```
Comment (Maheshkale447) [inline Ginger/Ginger/DevTimeTrackerIdle.cs:9]: Add documentation for this class

File: Ginger/Ginger/DevTimeTrackerIdle.cs [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+            idleTimer.Interval = TimeSpan.FromMinutes(1); // Check every minute
+            idleTimer.Tick += IdleTimer_Tick;
+            idleTimer.Start();
+
+            lastActivity = DateTime.Now;
+            isPaused = false;
+        }
+
+        public void AttachActivityHandlers(UIElement element)
+        {
```
Comment (Maheshkale447) [inline Ginger/Ginger/DevTimeTrackerIdle.cs:30]: Add a destructor method which will remove the attached events.

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]
Please review comments by Codacy and Code Rabbit

File: Ginger/Ginger/DevTimeTrackerIdle.cs [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+            if (isPaused)
+            {
+                ResumeDevelopmentTimeTracker();
+            }
+
+            lastActivity = DateTime.Now;
+
+        }
+        private List<RepositoryItemBase> _itemsWithPausedDevelopmentTimeTracker = [];
```
[the remark this answers is not in the document]
Comment (AmanPrasad43) [reply Ginger/Ginger/DevTimeTrackerIdle.cs:55]: this a new way for declaration.

**Review by Maheshkale447 — APPROVED** [review]

**Pull request merged**